fornclake/TetraForce
# Added identity system & Itch launcher support

![image](https://user-images.githubusercontent.com/21203171/138635413-1601f0a2-8b83-4c59-a7d5-b54f561d5251.png)
![image](https://user-images.githubusercontent.com/21203171/138635428-f444b553-5857-4713-b0db-6a7db3084ff5.png)


### Summary
- Added custom identity service addon to game
- Itch identity handler
- Server identity validation
- Added disabled LineEdit to theme
- Added tooltip hint to theme

Adding support for other identity system and launchers in the future will only require changes to the addon, unless a new UI element is needed, it handles the rest.

### Testing
You can connect to a multiplayer game the traditional way to validate nothing broke, but testing Itch support is not super straightforward, as TetraForce does not having an existing Itch page...

Here are the steps:
1. Create a build of TetraForce named `TetraForce.exe`
2. Copy `/.itch/windows.toml` to the same directory and rename it `.itch.toml`
3. Zip up `TetraForce.exe`, `TetraForce.pck`, and `.itch.toml` into a `zip` file
4. Create a new private Itch game
5. Upload `zip` and make sure to set it to Windows
6. Start itch launcher (install if not done so already)
7. Install game using Itch launcher (if it doesn't show your build it may take up to five minutes to update)
8. Launch game!

[**Has this been tested in multiplayer?**](https://github.com/loudsmilestudios/TetraForce/wiki/How-to-test-multiplayer) *yes*

---

I also think it would be cool to replace the `itch:` tag with an itch icon. I think it would be more pretty. So, I suggest a followup task "Replace platform name tag with icon"
